src/components/DpAddonWrapper/DpAddonWrapper.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+  <div>
+    <component
+      :is="component"
+      :ref="refComponent"
+      v-bind="addonProps"
+    />
+  </div>
+</template>
+
+<script>
+import { checkResponse, dpRpc } from '../../lib/DpApi'
+
+export default {
+  name: 'AddonWrapper',
+
+  props: {
+    /**
+     * The addonProps prop will be binded to the addon component to add props dynamically.
+     */
+    addonProps: {
+      type: Object,
+      required: false,
+      default: () => {}
+    },
+
+    /**
+     * The hookName prop will be used to load an addon via the generic rpc route.
+     */
+    hookName: {
+      type: String,
+      required: true,
+      default: ''
+    },
+
+    refComponent: {
+      type: String,
+      required: false,
+      default: ''
+    }
+  },
+
+  data () {
+    return {
+      component: ''
+    }
+  },
+
+  methods: {
+    loadComponents () {
+      dpRpc('addons.assets.load', { hookName: this.hookName })
+        .then(response => checkResponse(response))
+        .then(response => {
+          const result = response[0].result
+
+          for (const key of Object.keys(result)) {
+            const addon = result[key]
+            if (addon === undefined) {
+              /*
+               * If for some reason we don't receive a valid info object from the backend
```

Comment:
What is an "info object"?

Reply:
it should be ` a valid response..` sorry I just copy pasted this component before. Done here: https://github.com/demos-europe/demosplan-ui/pull/473/commits/88306d827a131b1bd75c029c4286b79dd5ddfbff